src/content-helper/dashboard-page/pages/traffic-boost/preview/hooks/use-iframe-highlight.ts

```diff
@@ -0,0 +1,601 @@
+				highlightSpan.className = isPrevious
+					? `${ className } previous-suggestion`
+					: className;
+
+				// Find if the range is within a link and if it encompasses the entire link text.
+				const container = range.commonAncestorContainer;
+				const linkNode = container.nodeType === Node.ELEMENT_NODE
+					? ( container as Element ).closest( 'a' )
+					: ( container as Node ).parentElement?.closest( 'a' );
+
+				const isFullLinkSelected = linkNode && range.toString().trim() === linkNode.textContent?.trim();
+
+				if ( isFullLinkSelected && linkNode ) {
+					// Create a new span and insert it before the link.
+					linkNode.parentNode?.insertBefore( highlightSpan, linkNode );
+
+					// Move the link into the span.
+					highlightSpan.appendChild( linkNode );
+				} else {
+					// Normal case - no links or partial link selection.
+					range.deleteContents();
+					highlightSpan.appendChild( fragment );
+					range.insertNode( highlightSpan );
+				}
+
+				return highlightSpan;
+			} catch ( e ) {
+				// eslint-disable-next-line no-console
+				console.error( 'WP Parsely: Error highlighting range', e );
+			}
+		}, [ iframeRef ] );
+
+	/**
+	 * Removes highlight spans from the iframe content.
+	 *
+	 * @since 3.18.0
+	 *
+	 * @param {HTMLIFrameElement} iframe        The iframe element to remove highlights from.
+	 * @param {string}            querySelector The query selector to find highlight elements.
+	 * @param {boolean}           animate       Whether to animate the removal of highlights.
+	 */
+	const removeHighlights = useCallback( ( iframe: HTMLIFrameElement, querySelector: string, animate = false ) => {
+		try {
+			const iframeDocument = iframe.contentDocument ?? iframe.contentWindow?.document;
+			if ( ! iframeDocument ) {
+				return;
+			}
+
+			const removeAndClean = ( highlight: Element, parent: ParentNode ) => {
+				// Move all child nodes before the highlight span.
+				while ( highlight.firstChild ) {
+					parent.insertBefore( highlight.firstChild, highlight );
+				}
+				parent.removeChild( highlight );
+				parent.normalize();
+
+				// Remove any anchors without text in the parent node.
+				const anchors = parent.querySelectorAll( 'a' );
+				anchors.forEach( ( anchor ) => {
+					if ( ! anchor.textContent ) {
+						parent.removeChild( anchor );
+						return;
+					}
+
+					// Check if the adjacent anchor has the same href or smartlink attribute.
+					const nextAnchor = anchor.nextElementSibling as HTMLAnchorElement;
+					if ( nextAnchor && (
+						anchor.href === nextAnchor.href ||
+						( anchor.getAttribute( 'data-smartlink' ) === nextAnchor.getAttribute( 'data-smartlink' ) &&
+						anchor.getAttribute( 'data-smartlink' ) !== null )
+					) ) {
+						// Merge the anchors.
+						while ( nextAnchor.firstChild ) {
+							anchor.appendChild( nextAnchor.firstChild );
+						}
+						parent.removeChild( nextAnchor );
+					}
+				} );
+			};
+
+			// Function to recursively unwrap nested highlights.
```

Comment:
Maybe we should have a function DocBlock for this one, like for other constants which are functions?